packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/AmplifyPushNotificationsPlugin.kt

```diff
@@ -60,53 +66,54 @@ open class AmplifyPushNotificationsPlugin : FlutterPlugin, ActivityAware,
      */
     private var mainBinaryMessenger: BinaryMessenger? = null
 
-    /**
-     * Flutter API interface.
-     */
-    private var flutterApi: PushNotificationsFlutterApi? = null
 
     /**
      * Launch notification has the notification map when app was launched by tapping on the notification
      * when the app is in killed state and null otherwise.
      */
     private var launchNotification: MutableMap<Any, Any?>? = null
 
+
     override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
         mainBinaryMessenger = flutterPluginBinding.binaryMessenger
+        FlutterEngineCache.getInstance().put(
+            PushNotificationPluginConstants.FLUTTER_ENGINE_ID,
+            flutterPluginBinding.flutterEngine
```

Comment:
The underlying Java method [`getFlutterEngine()` is deprecated](https://api.flutter.dev/javadoc/io/flutter/embedding/engine/plugins/FlutterPlugin.FlutterPluginBinding.html#getFlutterEngine()), how would this work when Flutter finally drops it?

Reply:
Good question. Thinking about it, this is pretty much only to indicate that a Fluter Engine exists in the group. Perhaps the same can be done with a bool in Shared Prefs.

Reply:
OK if there is another way to do check, can add a TODO comment and update later.